src/filters/detectors/assurance.cpp

```diff
@@ -9,71 +10,174 @@
 namespace mesos {
 namespace serenity {
 
-Result<Detection> AssuranceDetector::processSample(
-    double_t in) {
-  this->window.push_back(in);
+void AssuranceDetector::shiftBasePoints() {
+  for (std::list<double_t>::iterator& basePoint : this->basePoints) {
+    basePoint++;
+  }
+}
+
 
-  if (this->window.size() < this->cfg.getU64(detector::WINDOW_SIZE)) {
-    return None();  // Only warm up.
+Detection AssuranceDetector::createContention(double_t severity = -1) {
+  Detection cpd;
+  if (severity > 0) {
+    cpd.severity = severity;
   }
 
-  double_t basePoint = this->window.front();
+  SERENITY_LOG(INFO) << " Created contention with severity = "
+    << (cpd.severity.isSome() ? std::to_string(cpd.severity.get()) : "<none>");
+  return cpd;
+}
+
+
+void AssuranceDetector::recalculateParams() {
+  this->window.clear();
+  this->basePoints.clear();
+
+  uint64_t windowSize = this->cfg.getU64(detector::WINDOW_SIZE);
```

Comment:
Don't call cfg object. If you need something that is inside it, please create a member in your class and copy value in constructor.
It will make it easier to maintain the class, because you will know what config values you use.